File: app/src/main/java/com/primoz/timetimer/realmHelper/DataHelper.java
/*
package com.primoz.timetimer.realmHelper;

import android.util.Log;

import com.primoz.timetimer.adapters.MyWorkoutsAdapter;
import com.primoz.timetimer.realmObjects.Workout;

import io.realm.Realm;

*/
/**
 * Created by Primož on 21/08/2017.
 *//*


public class DataHelper {

    public static void addItemAsync(Realm realm, final String name, final int timeTotal, final int timeWork, final int timeRest, final int timeRounds) {
        realm.executeTransactionAsync(new Realm.Transaction() {
                                          @Override
                                          public void execute(Realm realm) {
                                              Workout workout = realm.where(Workout.class).equalTo("name", name).findFirst();
                                              if (workout != null) {
                                                  workout.setName(name);
                                                  workout.setTimeTotal(timeTotal);
                                                  workout.setTimeWork(timeWork);
                                                  workout.setTimeRest(timeRest);
                                                  workout.setTimeRounds(timeRounds);
                                              } else {
                                                  Workout.createWorkout(realm, name, timeTotal, timeWork, timeRest, timeRounds);
                                              }                                          }
                                      }
        );
    }

    public static void addItem(Realm realm, final String oldName, final String name, final int timeTotal, final int timeWork, final int timeRest, final int timeRounds) {
        realm.executeTransaction(new Realm.Transaction() {
                                     @Override
                                     public void execute(Realm realm) {
                                         Workout workout = realm.where(Workout.class).equalTo("name", oldName).findFirst();
                                         if (workout != null) {
                                             workout.setName(name);
                                             workout.setTimeTotal(timeTotal);
                                             workout.setTimeWork(timeWork);
                                             workout.setTimeRest(timeRest);
                                             workout.setTimeRounds(timeRounds);
                                         } else {
                                             Workout.createWorkout(realm, name, timeTotal, timeWork, timeRest, timeRounds);
                                         }
                                     }
                                 }
        );
    }

    public static void deleteItemAsync(Realm realm, final String name, final MyWorkoutsAdapter adapter) {
        realm.executeTransactionAsync(new Realm.Transaction() {
            @Override
            public void execute(Realm realm) {
                Workout.delete(realm, name);
            }
        }, new Realm.Transaction.OnSuccess() {
            @Override
            public void onSuccess() {
                adapter.notifyDataSetChanged();
            }
        });
    }

}
*/

File: app/src/main/java/com/primoz/timetimer/data_mvp/source/DataHelper.kt
package com.primoz.timetimer.data_mvp.source

import com.primoz.timetimer.data_mvp.Workout
import com.primoz.timetimer.data_mvp.WorkoutPOJO
import com.primoz.timetimer.data_mvp.WorkoutsDataSource

import io.realm.Realm
import io.realm.kotlin.where

/**
 * Created by Primož on 21/08/2017.
 */

object DataHelper {

    fun addNewWorkout(realm: Realm, workoutPOJO: WorkoutPOJO, listener: WorkoutsDataSource.SaveWorkoutCallback? = null) {
        realm.executeTransaction { realmInstance ->
            val name = workoutPOJO.name
            val timeWork = workoutPOJO.timeWork
            val timeRest = workoutPOJO.timeRest
            val timeRounds = workoutPOJO.timeRounds

            Workout.createWorkout(realmInstance, name, timeWork, timeRest, timeRounds)
            listener?.onWorkoutSaved()
        }
    }


    fun editWorkout(realm: Realm, workoutID: Int, workoutPOJO: WorkoutPOJO, listener: WorkoutsDataSource.EditWorkoutCallback? = null) {
        realm.executeTransaction { realmInstance ->
            realmInstance.where<Workout>().equalTo("workoutID", workoutID).findFirst()?.let {
                it.name = workoutPOJO.name
                it.timeWork = workoutPOJO.timeWork
                it.timeRest = workoutPOJO.timeRest
                it.timeRounds = workoutPOJO.timeRounds
                listener?.onWorkoutEdited()
            }
        }
    }

    fun deleteItemAsync(realm: Realm, workoutID: Int) {
        realm.executeTransaction { realmInstance -> Workout.delete(realmInstance, workoutID) }
    }

    fun addOrEditWorkout(realm: Realm, title: String, work: Int, rest: Int, rounds: Int) {
        val workout = realm.where<Workout>().equalTo("name", title).findFirst()
        if (workout != null) { //Edit
            DataHelper.editWorkout(realm, workout.workoutID,
                    WorkoutPOJO(title,
                            work,
                            rest,
                            rounds))
        } else {//New
            DataHelper.addNewWorkout(realm, WorkoutPOJO(title,
                    work,
                    rest,
                    rounds))
        }
    }

}
